plugins/woocommerce/src/Internal/Admin/Homescreen.php

```diff
@@ -50,6 +51,82 @@ public function __construct() {
 			add_action( 'admin_head', array( $this, 'update_link_structure' ), 20 );
 		}
 		add_filter( 'woocommerce_admin_preload_options', array( $this, 'preload_options' ) );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'maybe_set_default_shipping_options_on_home' ), 9999 );
+	}
+
+	/**
+	 * Set free shipping in the same country as the store default
+	 * Flag rate in all other countries when any of the following conditions are ture
+	 *
+	 * - The store sells physical products, has JP and WCS installed and connected, and is located in the US.
+	 * - The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).
+	 * - The store sells physical products and is located in US, but JP and WCS are not installed.
+	 *
+	 * @param array $settings shared admin settings.
+	 * @return array
+	 */
+	public function maybe_set_default_shipping_options_on_home( $settings ) {
```

Comment:
@moon0326 Could you wrap this under a feature flag logic?  I've added a feature flag [`shipping-smart-defaults`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/config/core.json#L11).

Reply:
Yup! Just noticed the feature flag in your PR 👍 

I'll make an update.

Reply:
@chihsuan Updated in https://github.com/woocommerce/woocommerce/pull/33547/commits/82abea91f94b84fef02ba265576b7e346bf7d1e4